custom/icds/models.py

```diff
@@ -0,0 +1,148 @@
+
+from casexml.apps.case.models import CommCareCase
+from corehq.apps.userreports.expressions.getters import transform_date
+
+
+from .constants import *
+
+
+class ChildHealthCaseRow(object):
+    def __init__(self, case, snapshot_start_date, snapshot_end_date, forms_provider=[]):
+        """
+        Snapshot of the case in the duration specified by snapshot_start_date and snapshot_end_date
+
+        args:
+            case: case of type 'child_health'
+            snapshot_start_date: Start date of duration to snapshot the case on
+            snapshot_end_date: End date of duration to snapshot the case on.
+
+        dates can be date/datetime objects or a valid strings
+        """
+        if not isinstance(case, CommCareCase):
+            case = CommCareCase.wrap(case)
+        self.case = case
+        self.snapshot_start_date = transform_date(snapshot_start_date)
+        self.snapshot_end_date = transform_date(snapshot_end_date)
+        self.forms_provider = forms_provider
+
+    # needs caching
```

Comment:
Still figuring out this and planning to do this in a separate PR. Safe to not block on this, as these wont be used in the UCRs yet.